### ISSUE TITLE
How to use?

### ISSUE THREAD

**Issue by kepoorhampond** [gawlk/lamp#1]
I recently saw your post on r/unixporn and I though I'd like to give your setup a try, but I can't figure out how to use this script. I've got `art` running, but I don't think in the proper way to run lamp as well.

**Comment by gawlk** [comment]
It's in a quite early stage, that's why there is not much in the README yet.

So, the idea is to run lamp after each and every window change made via your window manager.
Here are two configurations that has it implemented: [bspwm's sxhkd config](https://github.com/gawlk/dots/blob/master/sxhkd/sxhkdrc) and [2bwm's config](https://github.com/gawlk/dots/blob/master/2bwm/config.h)

Let's take a look at `sxhkdrc` which is much smaller and easier to understand.
 `lamp` is ran after:
- opening a terminal
- opening a launcher
- killing a window
- changing the focused workspace
- changing the focused window

One huge issue that you'll have, is if you have the `pointer follow` option enabled in your WM. In `2bwm` it's quite hard to disable it and even a bit counter intuitive so I left it be but I had focused window that weren't be detected by `lamp`. So I had to [change the source code and add a line that launches `lamp`](https://github.com/gawlk/dots/blob/master/2bwm/2bwm.c?utf8=%E2%9C%93#L1632) whenever a new window is focused.

In other words, if you can, I really suggest you to disable the `pointer follow` option (if you can) since it'll save you from a lot of trouble. If not, either run away or good luck haha.

Also, `lamp` uses window titles to detect with terminal is which. So you'll need to add:
```bash
# Set title of the window (used im lamp)
PROMPT_COMMAND='echo -en "\033]0;$(tty)\007"'
```
to your .bashrc

Anyway, good luck !

**Comment by kepoorhampond** [comment]
Great, thank a lot!

**Comment by gawlk** [comment]
No problem ! 👍 

If you don't have any more questions about how to use it, please close the issue.
